### PR TITLE
core: full deletion by sender based on preference; don't overwrite item content on "mark deleted"

### PR DESCRIPTION
**Pull request by jr-simplex**

### CODE REVIEW

File: tests/ChatTests.hs [hunk 1]
```diff
@@ -4249,7 +4299,7 @@ itemId :: Int -> String
 itemId i = show $ length chatFeatures + i
 
 groupItemId :: Int -> Int -> String
-groupItemId n i = show $ (length chatFeatures) * n + i
+groupItemId n i = show $ length chatFeatures * n + i
```
Comment (jr-simplex) [inline tests/ChatTests.hs:4302]: why does it use `chatFeatures` and not `groupFeatures`?
```suggestion
groupItemId n i = show $ length groupFeatures * n + i
```

File: src/Simplex/Chat/Types.hs [hunk 2]
```diff
@@ -261,6 +261,11 @@ chatFeatureToText = \case
   CFFullDelete -> "Full deletion"
   CFVoice -> "Voice messages"
 
+featureAllowed :: (PrefEnabled -> Bool) -> Contact -> ChatFeature -> Bool
+featureAllowed forWhom Contact {mergedPreferences} feature =
```
Comment (epoberezkin) [inline src/Simplex/Chat/Types.hs:265]:
```suggestion
featureAllowed feature forWhom Contact {mergedPreferences} =
```

File: src/Simplex/Chat/View.hs [hunk 3]
```diff
@@ -334,19 +338,20 @@ viewItemUpdate chat ChatItem {chatDir, meta, content, quotedItem} ts = case chat
...
-      CIDMInternal -> ["message deleted"]
-    _ -> ["message deleted"]
-  GroupChat g -> case (chatDir, deletedContent, toContent) of
-    (CIGroupRcv GroupMember {localDisplayName = m}, CIRcvMsgContent mc, CIRcvDeleted mode) -> case mode of
-      CIDMBroadcast -> viewReceivedMessage (ttyFromGroupDeleted g m) [] mc ts meta
-      CIDMInternal -> ["message deleted"]
-    _ -> ["message deleted"]
-  _ -> []
+viewItemDelete :: ChatInfo c -> ChatItem c d -> Bool -> Bool -> CurrentTime -> [StyledString]
+viewItemDelete chat ChatItem {chatDir, meta, content = deletedContent} markedDeleted byUser ts =
```
Comment (epoberezkin) [inline src/Simplex/Chat/View.hs:342]: use guard

File: src/Simplex/Chat/View.hs [hunk 4]
```diff
@@ -334,19 +338,20 @@ viewItemUpdate chat ChatItem {chatDir, meta, content, quotedItem} ts = case chat
...
+    else case chat of
+      DirectChat Contact {localDisplayName = c} -> case (chatDir, deletedContent) of
+        (CIDirectRcv, CIRcvMsgContent mc) -> viewReceivedMessage (ttyFromContactDeleted c markedDeleted) [] mc ts meta
+        _ -> prohibited
+      GroupChat g -> case (chatDir, deletedContent) of
+        (CIGroupRcv GroupMember {localDisplayName = m}, CIRcvMsgContent mc) -> viewReceivedMessage (ttyFromGroupDeleted g m markedDeleted) [] mc ts meta
+        _ -> prohibited
+      _ -> prohibited
+  where
+    prohibited = [styled (colored Red) ("[prohibited message deletion happened, please report it to developers]" :: String)]
```
Comment (epoberezkin) [inline src/Simplex/Chat/View.hs:354]:
```suggestion
    prohibited = [styled (colored Red) ("[unexpected message deletion, report to developers]" :: String)]
```

File: tests/ChatTests.hs (hunk 1 above)
Comment (epoberezkin) [reply tests/ChatTests.hs:4302]: ah - that's why the test was failing... :)

Comment (epoberezkin) [reply tests/ChatTests.hs:4302]: check out the change in my direct messages commit - I just changed numbers there incorrectly then.

File: src/Simplex/Chat/View.hs [hunk 5]
```diff
@@ -312,7 +316,7 @@ viewChatItem chat ChatItem {chatDir, meta, content, quotedItem, file} doShow ts
     showRcvItemProhibited from = showItem $ receivedWithTime_ ts from [] meta [plainContent content <> " " <> prohibited]
     showItem ss = if doShow then ss else []
     plainContent = plain . ciContentToText
-    prohibited = styled (colored Red) ("[prohibited - it's a bug if this chat item was created in this context, please report it to dev team]" :: String)
+    prohibited = styled (colored Red) ("[prohibited - it's a bug if this chat item was created in this context, please report it to developers]" :: String)
```
Comment (epoberezkin) [inline src/Simplex/Chat/View.hs:319]:
```suggestion
    prohibited = styled (colored Red) ("[unexpected chat item created, report to developers]" :: String)
```

**Pull request merged**